modules/codegen/src/main/scala/com/twilio/guardrail/generators/syntax/Java.scala

```diff
@@ -214,13 +215,26 @@ object Java {
     defns.sortWith((a, b) => sortKeyFor(a) - sortKeyFor(b) < 0)
   }
 
-  lazy val SHOWER_CLASS_DEF: ClassOrInterfaceDeclaration = {
-    JavaParser
-      .parseResource(getClass.getClassLoader, "java/Shower.java", StandardCharsets.UTF_8)
-      .getClassByName("Shower")
-      .asScala
-      .getOrElse(
-        throw new AssertionError("Shower.java in class resources is not valid")
-      )
-  }
+  def loadSupportDefinitionFromResources(className: String): Target[SupportDefinition[JavaLanguage]] =
+    Try(JavaParser.parseResource(getClass.getClassLoader, s"java/${className}.java", StandardCharsets.UTF_8)) match {
+      case Failure(t) =>
+        Target.raiseError[SupportDefinition[JavaLanguage]](s"Failed to load and parse class ${className} from resources: $t")
+      case Success(cu) =>
+        cu.getClassByName(className)
+          .asScala
+          .fold(
+            Target.raiseError[SupportDefinition[JavaLanguage]](s"Unable to find class ${className} in parsed resource file")
+          )(
+            clsDef =>
+              Target.pure(
+                SupportDefinition[JavaLanguage](
+                  new Name(className),
+                  cu.getImports.toList,
+                  clsDef
+                )
+            )
+          )
+    }
+
+  lazy val SHOWER_SUPPORT_DEF: Target[SupportDefinition[JavaLanguage]] = loadSupportDefinitionFromResources("Shower")
```

Comment:
Is there any reason to have a singleton here? Why not `def`?

Reply:
It's intended to be used as an immutable object.  No reason aside from that.

Reply:
understood